client_test.go

```diff
@@ -208,3 +209,22 @@ func TestUseFstatChecked(t *testing.T) {
 	testFstatOption(t, UseFstat(true), true)
 	testFstatOption(t, UseFstat(false), false)
 }
+
+type sink struct{}
+
+func (*sink) Close() error                { return nil }
+func (*sink) Write(p []byte) (int, error) { return len(p), nil }
+
+func TestClientZeroLengthPacket(t *testing.T) {
+	// Packet length zero (never valid). This used to crash the client.
+	packet := []byte{0, 0, 0, 0}
+
+	r := bytes.NewReader(packet)
+	c, err := NewClientPipe(r, &sink{})
+	if err == nil || err == io.EOF {
```

Comment:
I‘m not so sure how useful it is to check that we didn’t get an `io.EOF` back.

Reply:
Well, it ties in with #390 and makes the test robust in the face of further changes to recvPacket. It's not essential and I can remove it if you want.

Reply:
I think here, we should just be fine that we get some sort of error, we really don’t care what here.

Reply:
Ok, changed to err == nil.